src/Symfony/Component/DependencyInjection/Compiler/AutowireRequiredMethodsPass.php

```diff
@@ -49,6 +50,14 @@ protected function processValue($value, bool $isRoot = false)
             }
 
             while (true) {
+                if (method_exists($r, 'getAttributes') && $r->getAttributes(Required::class)) {
+                    if ($this->isWither($r, $r->getDocComment() ?: '')) {
```

Comment:
btw, shouldn't isWither() return `true` when the `static` return type is used?

Reply:
Yes, that's kind of the point of that method (#36636).